### DISCUSSION TITLE
`igl::boundary_conditions`  for 2d meshes

### DISCUSSION THREAD

**Discussion by aditya18007** [libigl/libigl#1932] [Q&A] [answered]
I have been trying to use `igl::boundary_condition` for 2D meshes. I keep getting segfault at [this line](https://github.com/libigl/libigl/blob/main/include/igl/boundary_conditions.cpp#L85)

It appears to me that it is expecting 3d meshes only. Is there way to use the same for 2d meshes ?

**Comment by alecjacobson** [comment, marked as answer]
The bone-edges functionality has only been implemented for 3D. You could
find a 2D implementation here
https://github.com/alecjacobson/gptoolbox/blob/master/mesh/boundary_conditions.m

The offending line is just doing projection onto a line segment, so the 2D
version — while not implemented — is a straightforward analog.

(trivia: both gptoolbox and libigl grew out of my BBW project codebases,
hence the incredibly generic name for a rather specific set of boundary
conditions. These functions should probably be renamed).

>
